### PR TITLE
Fixes a ton of harddels

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request
Enabled by #57728

Cleans up a reference from all machines, anything which has mood, anything that effects the beauty of an area, and bar drones.
Becoming area sensitive involves adding yourself to your own area senstive list, and that list wasn't being
emptied, causing harddels

This appears to be what was causing non client related human harddels.

Fixes the radios, bot cores, and access cards of bots hard deling. qdel < QDEL_NULL
Fixes emmissive blockers harddeling, they were being ref'd by vis_contents
Fixes glass harddels properly, what the fuck why did you do this
Fixes an action related harddel. QDEL_NULL > qdel
Fixes sdql query harddels. The buttons just never gave up on their ref. WTF.
Fixes some soundloop harddels. I cry every time
Actually fixes some alien harddels, I blame robustin
Fixes some mob harddels, mostly bots. Quirky call to del() before finishing with the death proc chain my ass

Edit:There's too many, I can't keep this up to date. Check the commits for more details
## Why It's Good For The Game
I want more processing time for atmos god damn it

### CODE REVIEW

File: code/controllers/admin.dm [hunk 1]
```diff
@@ -5,10 +5,20 @@
 
 INITIALIZE_IMMEDIATE(/obj/effect/statclick)
 
-/obj/effect/statclick/Initialize(mapload, text, target) //Don't port this to Initialize it's too critical
+/obj/effect/statclick/Initialize(mapload, text, target) //Don't port this to Initialize it's too critical //They did it anyway
```
Comment (Mothblocks) [inline code/controllers/admin.dm:8]:
```suggestion
/obj/effect/statclick/Initialize(mapload, text, target)
```

Comments are for now, not for then.

File: code/controllers/admin.dm [hunk 2]
```diff
@@ -5,10 +5,20 @@
...
 	. = ..()
 	name = text
 	src.target = target
+	if(istype(target, /datum)) //Harddel man bad
+		RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/cleanup)
+
+/obj/effect/statclick/Destroy()
+	target = null
+	. = ..()
```
Comment (Mothblocks) [inline code/controllers/admin.dm:17]:
```suggestion
	return ..()
```

File: code/game/objects/items/devices/taperecorder.dm [hunk 3]
```diff
@@ -39,6 +39,12 @@
 	soundloop = new(list(src))
 	update_appearance()
 
+/obj/item/taperecorder/Destroy()
+	QDEL_NULL(soundloop)
+	if(mytape)
+		QDEL_NULL(mytape)
+	. = ..()
```
Comment (Mothblocks) [inline code/game/objects/items/devices/taperecorder.dm:46]:
```suggestion
	return ..()
```

File: code/game/objects/items/devices/taperecorder.dm [hunk 4]
```diff
@@ -39,6 +39,12 @@
 	soundloop = new(list(src))
 	update_appearance()
 
+/obj/item/taperecorder/Destroy()
+	QDEL_NULL(soundloop)
+	if(mytape)
+		QDEL_NULL(mytape)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/taperecorder.dm:45]:
```suggestion
	QDEL_NULL(mytape)
```

`qdel(null)` is a no-op, it's fine.

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 5]
```diff
@@ -41,6 +41,10 @@ GLOBAL_LIST_INIT(oilfry_blacklisted_items, typecacheof(list(
 	reagents.add_reagent(/datum/reagent/consumable/cooking_oil, 25)
 	fry_loop = new(list(src), FALSE)
 
+/obj/machinery/deepfryer/Destroy()
+	QDEL_NULL(fry_loop)
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:46]:
```suggestion
	return ..()
```

File: code/modules/food_and_drinks/kitchen_machinery/microwave.dm [hunk 6]
```diff
@@ -42,6 +42,8 @@
 	eject()
 	if(wires)
 		QDEL_NULL(wires)
+	if(soundloop)
+		QDEL_NULL(soundloop)
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/kitchen_machinery/microwave.dm:46]:
```suggestion
	QDEL_NULL(soundloop)
```

File: code/modules/spells/spell.dm [hunk 7]
```diff
@@ -38,7 +38,7 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell)) //needed for th
 
 /obj/effect/proc_holder/Destroy()
 	if (action)
-		qdel(action)
+		QDEL_NULL(action)
```
Comment (Mothblocks) [inline code/modules/spells/spell.dm:41]: Same thing here.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/controllers/admin.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/controllers/admin.dm:8]: ~~but it's funny~~

**Comment by LemonInTheDark** [comment]
I'm sorry for all the commits, I got a bit stuck in.
I'm done with new fixes here, I won't make reviewing this any more of a headache then it already is

**Comment by LemonInTheDark** [comment]
If it's any consolation, there's like one roundstart harddel left after all this, and I don't think there are many standard ones left either, at least not ones that can come up outside of player interaction

File: code/game/objects/items/devices/beacon.dm [hunk 8]
```diff
@@ -17,7 +17,7 @@
 		icon_state = "beacon-off"
 
 /obj/item/beacon/Destroy()
-	GLOB.teleportbeacons.Remove(src)
+	GLOB.teleportbeacons -= (src)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/beacon.dm:20]:
```suggestion
	GLOB.teleportbeacons -= src
```

File: code/game/objects/items/devices/beacon.dm [hunk 9]
```diff
@@ -27,7 +27,7 @@
 		GLOB.teleportbeacons += src
 	else
 		icon_state = "beacon-off"
-		GLOB.teleportbeacons.Remove(src)
+		GLOB.teleportbeacons -= (src)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/beacon.dm:30]:
```suggestion
		GLOB.teleportbeacons -= src
```

File: code/game/objects/items/tanks/jetpack.dm [hunk 10]
```diff
@@ -20,6 +20,10 @@
 	ion_trail.auto_process = FALSE
 	ion_trail.set_up(src)
 
+/obj/item/tank/jetpack/Destroy()
+	QDEL_NULL(ion_trail)
+	. = ..()
```
Comment (Mothblocks) [inline code/game/objects/items/tanks/jetpack.dm:25]:
```suggestion
	return ..()
```

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 11]
```diff
@@ -160,7 +160,7 @@
 
 /obj/structure/closet/crate/freezer/Destroy()
 	recursive_organ_check(src)
-	..()
+	. = ..()
```
Comment (Mothblocks) [inline code/game/objects/structures/crates_lockers/crates.dm:163]:
```suggestion
	return ..()
```

File: code/modules/cargo/orderconsole.dm [hunk 12]
```diff
@@ -41,7 +41,7 @@
 
 /obj/machinery/computer/cargo/Destroy()
 	QDEL_NULL(radio)
-	..()
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/cargo/orderconsole.dm:44]:
```suggestion
	return ..()
```

File: code/modules/clothing/gloves/tacklers.dm [hunk 13]
```diff
@@ -23,6 +23,10 @@
 	/// See: [/datum/component/tackler/var/skill_mod]
 	var/skill_mod = 0
 
+/obj/item/clothing/gloves/tackler/Destroy()
+	tackler = null
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/clothing/gloves/tacklers.dm:28]:
```suggestion
	return ..()
```

File: code/modules/food_and_drinks/kitchen_machinery/grill.dm [hunk 14]
```diff
@@ -20,6 +20,10 @@
 	. = ..()
 	grill_loop = new(list(src), FALSE)
 
+/obj/machinery/grill/Destroy()
+	QDEL_NULL(grill_loop)
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/kitchen_machinery/grill.dm:25]:
```suggestion
	return ..()
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Last round.

**Pull request merged**